File: Inventor-Training-Material-0c232f96f36e1e0bed31b07a6a6ebb901a31dd40/Module 10 - Events/Labs Solution - Events.C#/AddInEvents/Form1.cs
// VBConversions Note: VB project level imports
using System.Collections;
using System;
using Microsoft.VisualBasic;
using System.Diagnostics;
// End of VB project level imports

using Inventor;


namespace AddInEvents
{
	public partial class Form1
	{
		
		// This class is used to wrap a Win32 hWnd as a .Net IWind32Window class.
		// This is primarily used for parenting a dialog to the Inventor window.
		//
		// For example:
		// myForm.Show(New WindowWrapper(m_inventorApplication.MainFrameHWND))
		//
		private class WindowWrapper : System.Windows.Forms.IWin32Window
		{
			
			public WindowWrapper(IntPtr handle)
			{
				_hwnd = handle;
			}
			
			public IntPtr Handle
			{
				get
				{
					return _hwnd;
				}
			}
			
			private IntPtr _hwnd;
		}
		
		
		private Inventor.Application mApplication;
		private CSimpleInteraction mSimpleInterac;
		
		public Form1(Inventor.Application oApplication)
		{
			
			// This call is required by the Windows Form Designer.
			InitializeComponent();
			
			// Add any initialization after the InitializeComponent() call.
			mApplication = oApplication;
			
		}
		
		public void ShowModeless()
		{
			base.Show(new WindowWrapper((IntPtr) mApplication.MainFrameHWND));
		}
		
		public void Button1_Click(System.Object sender, System.EventArgs e)
		{
			
			if (mApplication.ActiveDocument != null)
			{
				mSimpleInterac = new CSimpleInteraction(mApplication);
			}
			
		}
		
	}
}

File: Inventor-Training-Material-0c232f96f36e1e0bed31b07a6a6ebb901a31dd40/Module 10 - Events/Labs Solution - Events.VB.Net/AddInEvents/Form1.vb
Imports Inventor

Public Class Form1

    ' This class is used to wrap a Win32 hWnd as a .Net IWind32Window class.
    ' This is primarily used for parenting a dialog to the Inventor window.
    '
    ' For example:
    ' myForm.Show(New WindowWrapper(m_inventorApplication.MainFrameHWND))
    '
    Private Class WindowWrapper
        Implements System.Windows.Forms.IWin32Window

        Public Sub New(ByVal handle As IntPtr)
            _hwnd = handle
        End Sub

        Public ReadOnly Property Handle() As IntPtr _
          Implements System.Windows.Forms.IWin32Window.Handle
            Get
                Return _hwnd
            End Get
        End Property

        Private _hwnd As IntPtr
    End Class


    Private mApplication As Inventor.Application
    Private mSimpleInterac As CSimpleInteraction

    Public Sub New(ByVal oApplication As Inventor.Application)

        ' This call is required by the Windows Form Designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.
        mApplication = oApplication

    End Sub

    Public Sub ShowModeless()
        MyBase.Show(New WindowWrapper(mApplication.MainFrameHWND))
    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click

        If (Not mApplication.ActiveDocument Is Nothing) Then
            mSimpleInterac = New CSimpleInteraction(mApplication)
        End If

    End Sub

End Class